src/plugins/intel_cpu/src/cpu_tensor.cpp

```diff
@@ -0,0 +1,103 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <memory>
+#include <string>
+#include <map>
+
+#include "cpu_tensor.h"
+#include "ie_blob.h"
+#include "ie_ngraph_utils.hpp"
+#include "openvino/runtime/iremote_tensor.hpp"
+#include "openvino/runtime/properties.hpp"
+
+namespace ov {
+namespace intel_cpu {
+
+//For the beginning probably one contructor should be enough
+Tensor::Tensor(MemoryPtr memptr) : m_memptr{memptr} {
+    OPENVINO_ASSERT(m_memptr != nullptr);
+
+    // only support plain data format ncsp.
+    auto memdesc = m_memptr->getDescPtr();
+    OPENVINO_ASSERT(memdesc->hasLayoutType(LayoutType::ncsp), "intel_cpu::Tensor only supports memory with ncsp layout.");
+
+    m_element_type = InferenceEngine::details::convertPrecision(memdesc->getPrecision());
+}
+
+void Tensor::set_shape(ov::Shape new_shape) { // this is not as simple as it looks
+// if the ov::Shape is defined, then call cloneWithNewDims to the Memory object memory desc
+// if ov::Shape is undefined (dynamic), we probably have to reintialize the internal Memory object
+// using plain memory format
+    auto desc = m_memptr->getDescPtr();
+    const auto newdesc = desc->cloneWithNewDims(new_shape, true);
+    m_memptr->redefineDesc(newdesc);
+}
+
+const ov::element::Type& Tensor::get_element_type() const { // trivial -> simply convert the precision type
+    return m_element_type;
+}
+
+const ov::Shape& Tensor::get_shape() const { //trivial -> convert cpu Shape to ov::Shape
+    auto& shape = m_memptr->getDescPtr()->getShape();
+    OPENVINO_ASSERT(shape.isStatic(), "intel_cpu::Tensor has dynamic shape.");
+
+    std::lock_guard<std::mutex> guard(m_lock);
+    m_shape = shape.toPartialShape().to_shape();
+    return m_shape;
+}
+
+size_t Tensor::get_size() const { // trivial -> call to getElementsCount to Shape
+    auto& desc = m_memptr->getDesc();
+    return desc.getShape().getElementsCount();
+}
+
+size_t Tensor::get_byte_size() const { // trivial-> call getCurrentMemSize to memory desc
+    auto& desc = m_memptr->getDesc();
+    return desc.getCurrentMemSize();
+}
+
+const ov::Strides& Tensor::get_strides() const { // here I would ask you to investiage what solution may be better
+// in the case of BlockedMemory desc, we can retrive Strides, but  I'm not sure that the format is the same
+// also it is not clear what to do if the strides does not make sense (some tricky things like Sparced memory)
+// perhaps we are not allowed to wrap any cpu Memory except built over BlockedMemoryDesc
+    OPENVINO_ASSERT(m_memptr->getDescPtr()->isDefined(), "intel_cpu::Tensor requires memory with defined strides.");
+
+    std::lock_guard<std::mutex> guard(m_lock);
+    update_strides();
+    return m_strides;
+}
+
+void Tensor::update_strides() const {
+    auto blocked_desc = m_memptr->getDescWithType<BlockedMemoryDesc>();
```

Comment:
It is still beter to have null check, just in case the mem descriptor type  in the memory object has been completely changed by some reason.

Reply:
done